src/mlpack/tests/test_catch_tools.hpp

```diff
@@ -22,10 +22,18 @@
     REQUIRE(std::abs((R) - (L)) <= (E) * std::abs(R))
 
 // Check the values of two matrices.
-inline void CheckMatrices(const arma::mat& a,
-                          const arma::mat& b,
+template <typename MatTypeA, typename MatTypeB,
+    typename = std::enable_if_t<arma::is_arma_type<MatTypeA>::value
+        && arma::is_arma_type<MatTypeB>::value
+        && std::is_same<typename MatTypeA::elem_type, typename MatTypeB::elem_type>::value
+        && !std::is_integral<typename MatTypeA::elem_type>::value
+        && !std::is_integral<typename MatTypeB::elem_type>::value>>
```

Comment:
Technically this last check (on `MatTypeB::elem_type`) isn't needed since we already know that `MatTypeA::elem_type` and `MatTypeB::elem_type` are the same.